### PR TITLE
Improve refund UX

### PR DESCRIPTION
**Pull request by Tomanhez**

### CODE REVIEW

File: tests/Behat/Page/Admin/OrderRefundsPage.php [hunk 1]
```diff
@@ -147,8 +157,29 @@ private function getOrderShipment(): NodeElement
...
+        }
+        return $element->getText();
+    }
+
     private function isRefundable(NodeElement $element): bool
     {
-        return !$element->find('css', '.checkbox')->hasClass('disabled');
+        return $this->getUnitRefundedTotal($element)!==$this->unitTotal($element);
+    }
+    private function getUnitRefundedTotal(NodeElement $orderShipment): string
```
Comment (lchrusciel) [inline tests/Behat/Page/Admin/OrderRefundsPage.php]:
```suggestion

    private function getUnitRefundedTotal(NodeElement $orderShipment): string
```

File: src/Resources/views/orderRefunds.html.twig [hunk 2]
```diff
@@ -111,49 +110,63 @@
...
-                        <textarea rows="3" name="sylius_refund_comment" id="sylius-refund-comment"></textarea>
+                    <div class="ui hidden divider"></div>
+                    <div class="ui stackable grid">
+                        <div class="two column row">
+                            <div class="ui form column">
+                                <div class="field">
+                                    <label for="payment-methods">{{ 'sylius.ui.payment_method'|trans }}</label>
+                                    <select id="payment-methods" name="sylius_refund_payment_method" class="ui fluid selection dropdown">
+                                        {% for payment_method in payment_methods %}
+                                        <option value="{{ payment_method.id }}" {{ (payment_method.code == original_payment_method.code) ? 'selected="selected"' : '' }}>{{ payment_method.name }}</option>
```
Comment (lchrusciel) [inline src/Resources/views/orderRefunds.html.twig]:
```suggestion
                                        	<option value="{{ payment_method.id }}" {{ (payment_method.code == original_payment_method.code) ? 'selected="selected"' : '' }}>{{ payment_method.name }}</option>
```

File: tests/Behat/Page/Admin/OrderRefundsPage.php [hunk 3]
```diff
@@ -35,7 +35,16 @@ public function pickUnitWithProductToRefund(string $productName, int $unitNumber
...
         $units = $this->getUnitsWithProduct($productName);
 
-        $units[$unitNumber]->find('css', '.checkbox input')->check();
+        $value = $units[$unitNumber]->find('css','td:nth-child(2)');
+
+        $refunded = substr($this->getUnitRefundedTotal($value), 1);
+
+        $total = substr($this->getUnitTotal($value), 1);
+
+        /** @var double $total */
```
Comment (Zales0123) [inline tests/Behat/Page/Admin/OrderRefundsPage.php]: Is it needed?

File: tests/Behat/Page/Admin/OrderRefundsPage.php [hunk 4]
```diff
@@ -47,14 +56,22 @@ public function pickPartOfUnitWithProductToRefund(string $productName, int $unit
...
     }
 
     public function pickOrderShipment(): void
     {
         $orderShipment = $this->getOrderShipment();
 
-        $orderShipment->find('css', '.checkbox input')->check();
+        $total = $this->getUnitTotal($orderShipment);
+
+        if($this->getUnitRefundedTotal($orderShipment) !== null) {
```
Comment (Zales0123) [inline tests/Behat/Page/Admin/OrderRefundsPage.php]: Missing space after `if`

File: tests/Behat/Page/Admin/OrderRefundsPage.php [hunk 5]
```diff
@@ -147,8 +164,30 @@ private function getOrderShipment(): NodeElement
         return $this->getDocument()->find('css', '#refunds .shipment');
     }
 
+    private function getTextFromElement(?NodeElement $element): string
+    {
+        if ($element === null) {
+            return '';
+        }
+        return $element->getText();
```
Comment (Zales0123) [inline tests/Behat/Page/Admin/OrderRefundsPage.php]: Missing blank line above

Comment (Zales0123) [reply tests/Behat/Page/Admin/OrderRefundsPage.php]: And it really can be `return ($element === null) ? '' : $element->getText();`

File: tests/Behat/Page/Admin/OrderRefundsPage.php [hunk 6]
```diff
@@ -147,8 +164,30 @@ private function getOrderShipment(): NodeElement
...
+    }
+
     private function isRefundable(NodeElement $element): bool
     {
-        return !$element->find('css', '.checkbox')->hasClass('disabled');
+        return $this->getUnitRefundedTotal($element)!==$this->getUnitTotal($element);
```
Comment (Zales0123) [inline tests/Behat/Page/Admin/OrderRefundsPage.php]: Missing spaces around `!==`

File: tests/Behat/Page/Admin/OrderRefundsPage.php [hunk 7]
```diff
@@ -147,8 +164,30 @@ private function getOrderShipment(): NodeElement
...
+    }
+
+    private function getUnitRefundedTotal(NodeElement $orderShipment): string
+    {
+        $unitRefundedTotal = $orderShipment->find('css','.unit-refunded-total');
```
Comment (Zales0123) [inline tests/Behat/Page/Admin/OrderRefundsPage.php]: Missing space after a comma

File: tests/Behat/Page/Admin/OrderRefundsPage.php [hunk 8]
```diff
@@ -147,8 +164,30 @@ private function getOrderShipment(): NodeElement
...
+
+        return $this->getTextFromElement($unitRefundedTotal);
+    }
+
+    private function getUnitTotal(NodeElement $orderShipment): string
+    {
+        $unitTotal = $orderShipment->find('css','.unit-total');
```
Comment (Zales0123) [inline tests/Behat/Page/Admin/OrderRefundsPage.php]: Same here

File: tests/Behat/Page/Admin/OrderRefundsPage.php (hunk 3 above)
Comment (Tomanhez) [reply tests/Behat/Page/Admin/OrderRefundsPage.php]: I think that if we get int and double variables, we lost the sum after the decimal point? I think well?

**Comment by GSadee** [comment]
Thanks, Tomasz! :tada:

**Pull request merged**